build_feature_tree.js

```diff
@@ -0,0 +1,143 @@
+const fs = require("fs");
+const path = require("path");
+const yaml = require("js-yaml");
+const _ = require("lodash");
+const utils = require("./lib/js/utils");
+
+const ASSET_PATH = path.resolve(__dirname, "application_features");
+const FEATURE_DATA = {};
+
+/**
+ * Search recursively for a key name in an object.
+ * @param {string} key
+ * @param {Object} object
+ * @returns {String[]} - List of object paths (usable with _.set / _.get)
+ * @todo Move to code.js
+ */
+const findKeys = (key, object) => {
+    const foundKeys = [];
+    const iterate = (obj, level) => {
+        Object.keys(obj).forEach((k) => {
+            const objPath = !level ? k : [level, k].join(".");
+            if (k === key) {
+                foundKeys.push(objPath);
+            } else if (typeof obj[k] === "object" && obj[k] !== null) {
+                iterate(obj[k], objPath);
+            }
+        });
+    };
+    iterate(object);
+    return foundKeys;
+};
+
+/**
+ * Patched version of lodash's _.get() which handles empty object paths.
+ * @param {Object} obj - The object to query
+ * @param {string} objPath - The path of the property to get
+ * @returns {*} - Resolved value
+ * @todo Move to code.js
+ */
+const getByPath = (obj, objPath) => {
```

Comment:
Maybe `getByPathOrUndefined` to clarify?

Reply:
I adjusted the docstring to give more details about the possible return values.